farm/evaluation/semantic_answer_similarity_evaluation.py

```diff
@@ -0,0 +1,109 @@
+    Returns per QA pair a) the similarity of the most likely prediction to all available gold labels
+                        b) the highest similarity of all predictions to gold labels
+                        c) the original input populated with semantic answer scores per prediction
+
+    :param result: Output of QAinferencer.inference_from_file when setting return_json=False. Contains predictions
+                   and annotation information.
+    :type result: List[QAPred]
+    :param sts_model_path_or_string: SentenceTransformers semantic textual similarity model, should be path or string
+                                     pointing to downloadable models.
+                                     Defaults to multilingual model = "paraphrase-xlm-r-multilingual-v1"
+                                     For more speed and English only, use "paraphrase-MiniLM-L6-v2"
+    :type str
+    :param debug: Collect additional info for interesting cases for debugging purposes (default False).
+                  Interesting cases are predictions that do not overlap with any of the gold labels (topnNAcc == 0).
+
+    :type debug: bool
+
+    Returns the average of the semantically evaluated best, and Top N predictions as well as the List of
+    QApreds (result) with the semantic eval score added to each prediction in its "meta" field.
+    :return best_pred_similarity, all_preds_highest_similarity, result, debug_info
+    """
+    text_preds, text_labels = _extract_QA_result(result)
+    assert len(text_preds) == len(text_labels)
+
+    model = SentenceTransformer(sts_model_path_or_string)
+    # Flatten predictions and labels into one list to improve encoding speed through proper batching
+    lengths = []
+    all_texts = []
+    for p, l in zip(text_preds, text_labels):
+        # TODO potentially exclude (near) exact matches from computations
+        all_texts.extend(p)
+        all_texts.extend(l)
+        lengths.append((len(p), len(l)))
+    embeddings = model.encode(all_texts)
+
+    # Compute similarities
+    top1_sim = []
+    topn_sim = []
+    debug_info = []
+
+    current_position = 0
+    for i, (len_p, len_l) in enumerate(lengths):
+        pred_embeddings = embeddings[current_position:current_position + len_p, :]
+        current_position += len_p
+        label_embeddings = embeddings[current_position:current_position + len_l, :]
+        current_position += len_l
+        sims = cosine_similarity(pred_embeddings, label_embeddings)
+        top1_sim.append(np.max(sims[0, :]))
+        topn_sim.append(np.max(sims))
+        for j,p in enumerate(result[i].prediction):
+            p.meta = {"semantic_answer_score":np.max(sims[j, :])}
+
+        # Add debug information
+        if debug:
+            # Exclude no answer labels from debug info
+            if len(result[i].ground_truth_answer) == 0:
+                pass
+            else:
+                current_info = _add_debug_info(result=result, i=i, sims=sims)
```

Comment:
`current_info` needs to be defined outside `if debug:` so that we can always return `current_info`

Reply:
Would be a good catch but we return debug_info which is defined outside.

Reply:
Ah the error comes from within the helper function. Will fix it.